python/pdstools/adm/ADMDatamart.py

```diff
@@ -1898,13 +1913,26 @@ def _write_params_file(self, temp_dir, model_id, only_active_predictors):
             yaml.dump(params, f)
 
     def _run_quarto_command(
-        self, temp_dir, qmd_file, output_type, output_filename, verbose
-    ):
+        self,
+        log_file: Path,
+        temp_dir: Path,
+        qmd_file: str,
+        output_type: str,
+        output_filename: str,
+        save_log_file: bool,
+    ) -> int:
         """Run the Quarto command to generate the report."""
+
+        def log_and_write(message: str, level: str = "INFO"):
```

Comment:
Honestly I would try to use OOTB logging as much as you can, simply saying `logger.info()`. You can then intercept your logger in the streamlit app or wherever else to capture/intercept and write to a file?

Reply:
Don't want to be a pain here, but the more custom the `log_and_write` function becomes, the further away it becomes from 'recognizable' python code. That's something I want to watch out for.

Reply:
As far as I know, logger handlers are at an application level, not a script level. So the convention is to simply use OOTB loggers on the script level (namely, the getLogger(__name__) object), so that consumers can decide how to read these logs. 

Streamlit is one of those application/consumers, so in the streamlit app itself we could then create such a logging stream handler to capture logs and put into a text file. Therefore, I wouldn't expect this code in the ADMDatamart.py file? Perhaps I'm misunderstanding though, so please feel free to correct me.